modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -1110,6 +1110,14 @@ public static String getParentName(ComposedSchema composedSchema, Map<String, Sc
         int nullSchemaChildrenCount = 0;
         boolean hasAmbiguousParents = false;
         List<String> refedWithoutDiscriminator = new ArrayList<>();
+        String schemaName = "";
```

Comment:
What is the point of this for loop? Why can't you just do `composedSchema.getName()` to get this schema's name? I think this needs a comment because it could be a pretty large performance impact on huge schema documents.

Reply:
This for loop is needed because the Schema.name is null for all schemas in allSchemas.
Do you want me to edit the getSchemas method to set the names in the returned schemas?
Use composedSchema.getName() like you have described.

Reply:
I see. No need to change, as I suspect ModelUtils will change pretty heavily by 5.0 anyway.